### PR TITLE
iOS Xcode -pthread fix / iOS optimisation level default

### PR DESCRIPTION
**Pull request by danoli3**
## pthread issue
- Same issue as fixed in https://github.com/openframeworks/openFrameworks/pull/3939

> this was spilling a ton of compiler warnings in osx - one for nearly every .cpp file.
> with threads moving over to std::thread, pthread is no longer in use.
> this affects only the project files / templates, and the pre-generated project files for the project generator.

Silence this:
pthread

## iOS Optimisation level -Os for Release rather than -O3
Set the release optimisation to be inline with the default apple guideline for a release project for iOS (-Os).
This also brings the project in line with the built libraries for the most part as well.

## iOS Optimisation level -O0 for Debug builds.
- While the default for Xcode 5+ for debug is -Os, this was typically designed for Objective-C and Swift applications. While the later versions of Xcode to seem to get it almost correct. It would be wise to get this command in line with the rest of the framework at -O0 (no optimisations) for the Debug oF lib.
- The Absence of the GNU optimisation command so far was in the idea that the default in this case, was the -O0. See @memo's thoughts in https://github.com/openframeworks/openFrameworks/issues/1022

> Fastest, Smallest: Optimize for size. This setting enables all 'Faster' optimizations that do not typically increase code size. It also performs further optimizations designed to reduce code size. [-Os]

Screenshot of iOS+OFLib project:
optimisation

Optimisation consistency: https://github.com/openframeworks/openFrameworks/issues/1022
Optimisation levels for Statics: https://github.com/openframeworks/openFrameworks/issues/3619

### CODE REVIEW

**Comment by danoli3** [comment]
@openframeworks/ios how does this look?

**Comment by arturoc** [comment]
i think we should use -O0 for debug, it makes it's super tricky to debug with optimizations and really difficult to understand what's going on if you don't know that optimizations are on.

**Comment by bilderbuchi** [comment]
Isn't that exactly what he did?

File: libs/openFrameworksCompiled/project/ios/Debug.xcconfig [hunk 1]
```diff
@@ -5,8 +5,9 @@ GCC_ENABLE_SSE3_EXTENSIONS = YES
 GCC_ENABLE_SUPPLEMENTAL_SSE3_INSTRUCTIONS = YES
 GCC_INLINES_ARE_PRIVATE_EXTERN = NO
 GCC_MODEL_TUNING = G5
+GCC_OPTIMIZATION_LEVEL = 0
```
Comment (danoli3) [inline libs/openFrameworksCompiled/project/ios/Debug.xcconfig]: I wonder if this is causing the issue on older mac's to crash with the simulator (too much data?)

**Pull request merged**